sys/compat/freebsd64/freebsd64_misc.c

```diff
@@ -659,19 +662,22 @@ freebsd64_copyout_strings(struct image_params *imgp, uintcap_t *stack_base)
 	 */
 	if (execpath_len != 0) {
 		destp -= execpath_len;
-		imgp->execpathp = cheri_setbounds(destp, execpath_len);
-		error = copyout(imgp->execpath, destp, execpath_len);
+		destp = __builtin_align_down(destp, sizeof(uint64_t));
+		imgp->execpathp = (void * __capability)
+		    cheri_setbounds(destp, execpath_len);
```

Comment:
Maybe cher_setbounds_exact here as well? I think it should be safe with PATH_MAX=1024.